### ISSUE TITLE
Analysis hang

### ISSUE THREAD

**Issue by baloghadamsoftware** [typetools/checker-framework#4207]
Hello,

I was testing various Java analyzer tools on different projects used as test projects. One analyzer tool was the test project of another one :-) Last time Spotbugs was the test project for the Checker Framework.

I used the following config (in the build.gradle file):
```
  checkerFramework {
      checkers = [
      'org.checkerframework.checker.compilermsgs.CompilerMessagesChecker',
      'org.checkerframework.checker.fenum.FenumChecker',
      'org.checkerframework.checker.formatter.FormatterChecker',
      'org.checkerframework.checker.index.IndexChecker',
      'org.checkerframework.checker.interning.InterningChecker',
      'org.checkerframework.checker.lock.LockChecker',
      'org.checkerframework.checker.nullness.NullnessChecker',
      'org.checkerframework.checker.optional.OptionalChecker',
      'org.checkerframework.checker.propkey.PropertyKeyChecker',
      'org.checkerframework.checker.regex.RegexChecker',
      'org.checkerframework.checker.signature.SignatureChecker',
      'org.checkerframework.checker.signedness.SignednessChecker',
      'org.checkerframework.checker.tainting.TaintingChecker',
      'org.checkerframework.checker.units.UnitsChecker',
      'org.checkerframework.common.aliasing.AliasingChecker',
      'org.checkerframework.common.reflection.ClassValChecker',
      'org.checkerframework.common.reflection.MethodValChecker',
      'org.checkerframework.common.returnsreceiver.ReturnsReceiverChecker'
    ]
    extraJavacArgs = [
      '-Awarns'
    ]
    excludeTests = true
  }
```
Checker Framework version is 3.9.1, build command was `./gradlew spotbugs:compileJava`. Spotbugs commit id was `c37cee56387bedef9110cbc8004d59ff08b75d67`. However, the analysis did not finish at all, I had to kill it after 67 hours. When I commented out all the lines, and kept only one at a time then the analysis finished in a few minutes (max was 17 minutes, but average was 3-4 minutes per checker). The machine where I tested has 24 logical threads and 48G of physical memory.

**Comment by mernst** [comment]
I'm sorry you are having trouble.  Unfortunately, your bug report does not let others reproduce your problem.
 * The snippet you provided is not syntactic:  it lacks at least a `}`.
 * The snippet you provided is not sufficient:  it lacks `plugins` and `apply plugin`.
 * You didn't say what files to put the snippet in.

Could you please provide a way to reproduce your problem?
Please provide a set of commands that can be cut-and-pasted into a command shell to reproduce your problem.  These commands would start with cloning spotbugs.

**Comment by baloghadamsoftware** [comment]
Hello,

Thank you for your quick reply. Here are the steps to reproduce:

$ git clone https://github.com/spotbugs/spotbugs.git
$ cd spotbugs
$ git checkout c37cee56387bedef9110cbc8004d59ff08b75d67
$ cp _<location of attached build.gradle>_ .
$ ./gradlew spotbugs:compileJava

[build.zip](https://github.com/typetools/checker-framework/files/5887262/build.zip)

And I agree, it is nearly a duplicate of #4206, but there here lack of CalledMethodsChecker changes the symptom from heap overflow to a hang. The reason may be or may not be the same one, so I thought that it is better to report them separately.

Please tell me if you need additional information.

**Comment by baloghadamsoftware** [comment]
It seems that this is indeed a duplicate of #4206. The only difference is that here the heap overflow error does not terminate the whole process, but deadlocks it.

**Comment by mernst** [comment]
How do you know that the heap overflow deadlocks the process?  (That would reduce CPU usage to zero.)  I would have guessed that it is thrashing.

**Comment by mernst** [comment]
@baloghadamsoftware A ping:  how do you know it is deadlocked rather than thrashing?
Can this issue be closed, if it is resolved by increasing the heap size?  (In #4206 you gave the non-obvious method for doing that.)
